src/main/kotlin/org/rust/ide/core/overrideImplement/generateUtil.kt

```diff
@@ -87,13 +81,11 @@ fun insertNewTraitMembers(selected: Collection<RsTraitMemberChooserMember>, impl
             selected.mapNotNull { it.member as? RsConstant }
     )
     val lastMethodOrBrace = impl.functionList.lastOrNull() ?: impl.lbrace ?: return
-    runWriteAction {
-        impl.addRangeAfter(
-                templateImpl.lbrace?.nextSibling,
-                templateImpl.rbrace?.prevSibling,
-                lastMethodOrBrace
-        )
-    }
+    impl.addRangeAfter(
```

Comment:
I've got an exception while performing automatic trait implementation. It looks like its cause could be this `runWriteAction` removal? @slavam2605, @matklad, how do you think, should we put it back?

```
AWT events are not allowed inside write action: java.awt.event.InvocationEvent[INVOCATION_DEFAULT,runnable=com.intellij.openapi.util.ActionCallback$$Lambda$257/2117917341@55337b94,notifier=null,catchExceptions=false,when=1490370577201] on sun.lwawt.macosx.LWCToolkit@1ae77ddd
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.openapi.application.impl.NoSwingUnderWriteAction.lambda$watchForEvents$0(NoSwingUnderWriteAction.java:36)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:376)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
   ...
```